Marlin/src/HAL/DUE/HAL_SPI.cpp

```diff
@@ -510,7 +510,7 @@
         spiRxBlock = (pfnSpiRxBlock)spiRxBlockX;
         break;
       default:
-        spiDelayCyclesX4 = ((F_CPU) / 1000000) >> (6 - spiRate) << 2; // spiRate of 2 gives the maximum error with current CPU
+        spiDelayNS = (((F_CPU) / 1000000) >> (6 - spiRate) << 2) * 100; // spiRate of 2 gives the maximum error with current CPU
```

Comment:
If this is now nanoseconds, then presumably it should no longer involve `(F_CPU / 1000000)` but should use the nanosecond equivalent of the original calculation.

Reply:
Note that (F_CPU / 1000000) cycles == 1µs, aka 1000 nanoseconds.

Reply:
…thus:
```cpp
spiDelayNS = 4000 >> (6 - spiRate);  // spiRate of 2 gives the maximum error with current CPU
```

Reply:
Ah, I see. I suppose I could just `* 1000` then, not sure how I got 100; that would make the 400 above invalid, as well. Though I'm sure there's a much nicer bitmath way of doing it. I'll look at the other HALs.

Reply:
Ah, ok. Thanks!!

Reply:
Simple enough to follow, since:
```cpp
#define DELAY_NS(x) DELAY_CYCLES((x) * ((F_CPU) / 1000000UL) / 1000UL)          // floor
```